smarthint/loaders/productListingPage.ts

```diff
@@ -0,0 +1,177 @@
+import { ProductListingPage } from "../../commerce/types.ts";
+import { AppContext } from "../mod.ts";
+import {
+  getFilterParam,
+  getPaginationInfo,
+  getSortParam,
+  resolvePage,
+  toFilters,
+  toProduct,
+  toSortOption,
+} from "../utils/transform.ts";
+import { redirect } from "deco/mod.ts";
+import { getSessionCookie } from "../utils/getSession.ts";
+import { FilterProp, SearchSort, SHProduct } from "../utils/typings.ts";
+import { RuleType } from "./PLPBanners.ts";
+import { getCategoriesParam } from "./recommendations.ts";
+
+export interface Props {
+  /**
+   * @hide
+   */
+  term?: string;
+  /**
+   * @description Number of products that must be returned per page
+   */
+  size: number;
+  /**
+   * @hide
+   */
+  searchSort?: SearchSort;
+  /**
+   * @hide
+   */
+  filter?: FilterProp[];
+  /**
+   * @hide
+   */
+  from?: number;
+  /**
+   * @hide
+   */
+  rule?: string;
+  /**
+   * @hide
+   */
+  ruletype?: RuleType;
+  /**
+   * @hide
+   */
+  condition?: {
+    field?: string;
+    value?: string;
+    validation?: string;
+  };
+  /**
+   * @description if its a category page setup your store (VTEX,Wake,Shopify,etc) loader here
+   */
+  page?: ProductListingPage | null;
+}
+
+/**
+ * @title Smarthint Integration - Product List Page
+ * @description Product List Page
+ */
+const loader = async (
+  props: Props,
+  req: Request,
+  ctx: AppContext,
+): Promise<ProductListingPage | null> => {
+  const { api, shcode, cluster } = ctx;
+  const {
+    size = 12,
```

Comment:
shouldn't this be configurable?

Reply:
`size` came from props, and `shcode/cluster` came from app config